### ISSUE TITLE
Enrich proxy spans with filter data

### ISSUE THREAD

**Issue by jcchavezs** [envoyproxy/envoy#21959]
I was looking at envoy traces generated both when you run a filter and when you don’t and I see traces are pretty much the same except for the fact that when a filter blocks a request, no upstream_cluster/upstream_cluster.name is set (which makes sense). Also there is no clue on the duration of the filter execution (at least in traces) which keeps you blind about the duration of the filter execution. Finally, AFAIU filters have no way to access the current span to set attributes or annotations (e.g. modsecurity won’t tell you what rule caused the blocking).

In summary what I’d like to see as outcome would be:
- traces display filter duration either as annotations in the trace (subspans might be overkilling I think) filters can enrich span attributes
- Have an attribute about which filter blocked the request.

In the future I'd like it to be included in the WASM ABI too.

**Comment by wbpcode** [comment]
Sound great. But it would bring some additional overhead. 🤔

> Have an attribute about which filter blocked the request.

May be response flag or response detail string is better for this requirement?

**Comment by jcchavezs** [comment]
This issue has been closed by a bot with zero context about the status nor use case.

**Comment by phlax** [comment]
it was closed due to inactivity - i can reopen it now - but you will need to either implement this yourself, or find someone willing to do so, as well as get it approved
